pkg/didmethod/peer/did.go

```diff
@@ -30,7 +32,59 @@ func newDid(doc *did.Doc) (string, error) {
 	if doc.PublicKey == nil || doc.Authentication == nil {
 		return "", errors.New("the genesis version must include public keys and authentication")
 	}
+	numBasis, err := hashString(doc)
+	if err != nil {
+		return "", err
+	}
+	messageIdentifier := []string{peerPrefix, numAlgo, encAlgo, "-", numBasis}
+	peerDID := strings.Join(messageIdentifier, "")
+
+	return peerDID, nil
+}
+
+func validate(doc *did.Doc) error {
+	did := doc.ID
+	hashedNumBasis, err := validateDid(did)
+	if err != nil {
+		return err
+	}
+	//removing did to calculate the hash of genesis doc
+	doc.ID = ""
+	numBasis, err := hashString(doc)
+	if err != nil {
+		return err
+	}
+
+	if !(numBasis == hashedNumBasis) {
+		return errors.New("hash of the doc doesnt match the computed hash")
+	}
+	return nil
+}
+
+func validateDid(did string) (string, error) {
```

Comment:
This function is doing two things: it is validating the peer DID format, and it's also returning its `numBasis`. I suggest three things:

* Add godocs explaining what it does
* Extract the portion that returns the numBasis of a given peer DID into its own function and use that one

Reply:
added comments and splited the function as suggested